install.md

```diff
@@ -212,6 +212,65 @@ Building in a container is simpler, but more restrictive:
 $ make binary
 ```
 
+### Build and Install shell completion scripts
+
+Skopeo can create shell completion scripts for bash, zsh, fish and powershell.
+
+This places the scripts under the `completions` directory:
+```bash
+$ make completions
+```
+
+If you want to install the files to the system directories (on linux) use:
+```bash
+$ sudo make install-completions
+```
+
+By default the completion scripts will show extra descriptions as help:
+```
+$ skopeo l[TAB]
+list-tags  (List tags in the transport/repository specified by the SOURCE-IMAGE)  logout     (Logout of a container registry)
+login      (Login to a container registry)
+```
+
+To turn this of you can generate the scripts with the `skopeo completion --no-desc` option,
+see below for the individual shell instructions.
+
+#### BASH
+`bash-completion` has to be installed on the system.
+
+To load the completion script into the current session run:\
+`source <(skopeo completion bash)`
+
+To make it available for all bash sessions run:\
+`skopeo completion -f /etc/bash_completion.d/skopeo bash`
+
+
+#### ZSH
+Shell completion needs to be already enabled in the environment. The following can be executed:\
+`echo "autoload -U compinit; compinit" >> ~/.zshrc`
+
+To make it available for all zsh sessions run:\
+`skopeo completion -f "${fpath[1]}/_skopeo" zsh`
+
+Once the shell is reloaded the auto-completion should be working.
+
+
+#### FISH
+To load the completion script into the current session run:\
+`skopeo completion fish | source`
+
+To make it available for all fish sessions run:\
+`skopeo completion -f ~/.config/fish/completions/skopeo.fish fish`
+
+#### POWERSHELL
+To load the completion script into the current session run:
+`skopeo completion powershell | Out-String | Invoke-Expression`
+
+To make it available in all powershell sessions that a user has, write the
+completion output to a file and source that to the user's powershell profile.
+More information about profiles is available with **Get-Help about_Profiles**.
```

Comment:
Why all these individual commands?

`make install{-completions}` already puts the scripts into auto-loaded locations, so I’d expect all of it to work without any further action (on the next bash session restart).

Documenting how to load the completion within the current session feels potentially useful, but a bit too specific —  the users would probably benefit from a generic knowledge how to reload all completions into the session. (We also run into the inconsistency of hiding the `skopeo completion` subcommand and not having a man page, but documenting how to use it here.)

Isn’t documenting how to _individually_ write the config files via `completion -f` completely redundant with the `make install{-completions}` command?

---

I think this could all be just dropped; or, alternatively, _after_ the `make install`, add some note that completions will only work on new shell instances, and point at upstream *-completion documentation for how to reload updated/added completions into existing shells.

Reply:
So I agree the almost all users will not care since the package manger (or make install) should install the scripts for them.
However I think the are also cases where this does not apply:

- I do not know what the status of windows support is but powershell users would have to do it manually so they need documentation on how to do it. (Also you can use powershell on linux too)
- What if skopeo is distributed as single static binary. This seems to be common for many go applications. In this case the user has to manually load the script so they need instructions.

I think pointing at the upstream docs would be good enough. At least it would prevent us from having to manually update it when something changes.

Reply:
> What if skopeo is distributed as single static binary

Ugh. You are quite right, that could happen, and those users could still benefit from completions.

OTOH these tend to be users who don’t have root (making installation into `/etc/` or `/usr` unviable), or they want a static binary to run from a “we-don’t-care-what-distribution-it-is” container in some automated system. Actual interactive by users who just copy around binaries seems rarer — especially because we don’t provide such a binary.

BTW just copying `bin/skopeo` around basically doesn’t work for interactive use because Skopeo refuses to copy images without `policy.json` (or an opt-out CLI flag that needs to be typed every time). So… if anything, we could document the GNU standards-defined `make install DESTDIR=…` to create a complete set of files that {can, need to} be installed.

Reply:
Updated to just point at `skopeo completion --help`